### PR TITLE
Added 'max_pool1d' nn function to tensorflow frontend

### PR DESCRIPTION
**Pull request by Aditya-Komaravolu**
max_pool1d #4477

### CODE REVIEW

**Comment by Aditya-Komaravolu** [comment]
@djl11 can you skim through this pull request and suggest any changes if required?

File: ivy/functional/frontends/tensorflow/nn.py [hunk 1]
```diff
@@ -123,3 +123,14 @@ def batch_normalization(x, mean, variance, offset, scale, variance_epsilon, name
     return x * ivy.astype(inv, x.dtype, copy=False) + ivy.astype(
         offset - mean * inv if offset is not None else -mean * inv, x.dtype
     )
+
+
+def max_pool1d(input, ksize, strides, padding, data_format='NWC', name=None):
+    return ivy.max_pool1d(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/tensorflow/nn.py:129]: Hi @Aditya-Komaravolu 
Thank you for your time to contribute to Ivy 🙂
This function is not implemented on Ivy functional API if you want to implement this function you can do it compositionally following the frontend guide on compositions here [Compositions](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html#compositions)

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/tensorflow/nn.py (hunk 1 above)
Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Hi @hirwa-nshuti, thank you for your feedback. So this function should be transferred to general.py?

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: No, basically Ivy doesn't have `max_pool1d` function implemented. Implementing this function's behavior can be done using other ivy functions that's what the guide is explaining on how to implement frontend functions using compositions.

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Oh okay, I get it now. Can I tag you if I get stuck anywhere?

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Yeah no problem feel free to tag me🙂

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Hey @hirwa-nshuti, I am unable to find which existing function is a close substitute to max_pool. Can you help please?

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: If you're unable to implement the function compositionally I can recommend you to leave this task and take the one that is suitable for you.
Thanks

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: @hirwa-nshuti I still want to try. I have a doubt. 

Can I deploy a maxpool1d function at ivy/functional/backend/tensorflow/layes.py and then reference it to the ivy/functional/frontend/tensorflow/layers.py ? I guess that might sort out the issue. I tried a sample code on my laptop and it seems to work.

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: The thing is as we are currently welcoming the Ivy functional API extension suggestion. You can suggest it the function here before going straight away with implementations [Extend Ivy Functional API](https://github.com/unifyai/ivy/issues/3856)

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Sorry my bad.

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: And also if you want to implement it to be fully supported by Ivy you will need to implement it across all Ivy backends. Feel free to open a PR with the extension implementation.

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: image

This is what have tried.

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: > 

So should I have to move this function to the extensions.py file and then request for a PR?

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: > And also if you want to implement it to be fully supported by Ivy you will need to implement it across all Ivy backends. Feel free to open a PR with the extension implementation. You can refer to the extension guideline here [Ivy API Extensions](https://lets-unify.ai/ivy/contributing/4_open_tasks.html?highlight=extension#ivy-api-extensions)

So should I move this function to `extensions.py` and then request a new PR? Also, do I have to write tests for this new code?

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Yeah once you implement the function to be fully supported by all ivy backends you'll have to write tests for it.

Comment (Aditya-Komaravolu) [reply ivy/functional/frontends/tensorflow/nn.py:129]: @hirwa-nshuti I will get back to this function once I make a successful PR. I want to work on cosine_similarity func. Shall I close this PR?

Comment (hirwa-nshuti) [reply ivy/functional/frontends/tensorflow/nn.py:129]: Yeah you can close the PR no worries

**Pull request closed without merging**